build.gradle

```diff
@@ -140,6 +140,15 @@ shadowJar {
     classifier = null
 }
 
+task setVersion() {
+    if(project.hasProperty('newVersion')) {
```

Comment:
Is this `if` necessary? Task should fail if the property is missing, shouldn't it?

Reply:
Hey @dblock added `if` to make sure task wont fail if not passed `-PnewVersion=x.x.x-SNAPSHOT`, assuming this task will be called from the [workflow](https://github.com/opensearch-project/opensearch-build/pull/2291), which should have `-PnewVersion`.

Error removing if condition with passed in `-PnewVersion`
* What went wrong:
A problem occurred evaluating root project 'opensearch-job-scheduler'.
> Could not get unknown property 'newVersion' for task ':setVersion' of type org.gradle.api.DefaultTask.

I see both pros and cons removing the if condition, so open for the change.

Reply:
I think you want to use `onlyIf`:

```
task setVersion {
    onlyIf { project.hasProperty('newVersion') }
    ant...
    doLast {
        println "Setting version to ${project.getProperty('newVersion')}."
    }
}
```